SO_properties.py

```diff
@@ -636,12 +636,116 @@ def Tgas(self):
             return None
         return (self.gas_temperatures * (self.gas_masses / self.Mgas)).sum()
 
+    @lazy_property
+    def Tgas_cy_weighted(self):
+        if self.Ngas == 0:
+            return None
+        return (
+            self.gas_temperatures * (self.gas_compY.value / self.gas_compY.sum().value)
+        ).sum()
+
+    @lazy_property
+    def Tgas_cy_weighted_no_agn(self):
+        if self.Ngas == 0:
```

Comment:
This seems like it might be wrong because we're checking if the halo has any gas particles but then we're dividing by the sum over the no AGN gas selection, so we could get a division by zero. Maybe we need to add a Ngas_no_agn property somewhere?